src/uws_client.c

```diff
@@ -1233,6 +1233,11 @@ static void on_underlying_io_bytes_received(void* context, const unsigned char*
                             LogError("Bad status (%d) received in WebSocket Upgrade response", status_code);
                             ws_open_result_detailed.result = WS_OPEN_ERROR_BAD_RESPONSE_STATUS;
                             ws_open_result_detailed.code = status_code;
+
+                            // Include the original http message so the client can determine if it cares to garner data.
+                            // Like for a redirect.
+                            ws_open_result_detailed.buffer = uws_client->stream_buffer;
```

Comment:
I am not sure about the life cycle of stream_buffer. I assume that you know it is still valid when the error reaches transport.cpp.  Carbon right now uses ThreadService to sync all calles into azure-c-shared-lib so we don't have to worry about thread safe issue here.

Reply:
Would it also be possible to include the headers? Often times this contains important information for debugging on the server side (e.g. request IDs(

Reply:
It's the raw stream from the service, so it has the headers.

(The use case in the Speech SDK is to read the location header.)